src/poetry/installation/installer.py

```diff
@@ -249,7 +256,21 @@ def _do_install(self, local_repo: Repository) -> int:
                 self._io,
             )
 
-            ops = solver.solve(use_latest=self._whitelist).calculate_operations()
+            if (
```

Comment:
I'm not 100% on if this is the only way to get at the config, but this smells to me. I'm pretty sure you should be able to get at the `Config` object from here, or at least create one from a Factory if not.